src/bundle/Resources/public/js/scripts/admin.search.filters.js

```diff
@@ -120,62 +128,29 @@
 
         toggleDisabledStateOnApplyBtn();
     };
-    const dateConfig = {
-        formatDate: (date) => eZ.helpers.timezone.formatShortDateTime(date, null),
-    };
-    const checkSelectFieldsFilled = (modal) => {
-        const inputs = modal.querySelectorAll('.ez-date-select');
-        const isFilled = [...inputs].every((input) => !!doc.querySelector(input.dataset.targetSelector).value.trim());
-        const methodName = isFilled ? 'removeAttribute' : 'setAttribute';
+    const setSelectedDateRange = (selectedDates, dateString, instance) => {
+        const dateRange = instance.input.closest('.ez-date-range');
 
-        modal.querySelector('.ez-btn--select')[methodName]('disabled', !isFilled);
-    };
-    const setSelectedDateRange = (event) => {
-        const modal = event.target.closest('.ez-modal');
-        const startInput = modal.querySelector('.ez-date-select--start');
-        const targetStartInput = doc.querySelector(startInput.dataset.targetSelector);
-        const endInput = modal.querySelector('.ez-date-select--end');
-        const targetEndInput = doc.querySelector(endInput.dataset.targetSelector);
-        const startDate = parseInt(targetStartInput.value, 10);
-        const endDate = parseInt(targetEndInput.value, 10);
-        const datePeriod = endDate - startDate;
-        const secondsInDay = 60 * 60 * 24;
-        const days = datePeriod / secondsInDay;
-
-        doc.querySelector(modal.dataset.periodSelector).value = `P0Y0M${days}D`;
-        doc.querySelector(modal.dataset.endSelector).value = endDate;
+        if (selectedDates.length === 2) {
+            const startDate = parseDateUTC(selectedDates[0]);
+            const endDate = parseDateUTC(selectedDates[1]);
+            const days = (endDate - startDate) / 86400;
 
-        toggleDisabledStateOnApplyBtn();
-    };
-    const updateSourceInputValue = (sourceInput, date) => {
-        if (!date.length) {
-            sourceInput.value = '';
-            sourceInput.dispatchEvent(event);
-
-            return;
+            doc.querySelector(dateRange.dataset.periodSelector).value = `P0Y0M${days}D`;
+            doc.querySelector(dateRange.dataset.endSelector).value = endDate;
+            applyBtn.removeAttribute('disabled', false);
+        } else {
+            doc.querySelector(dateRange.dataset.periodSelector).value = '';
+            doc.querySelector(dateRange.dataset.endSelector).value = 0;
```

Comment:
Why do you have to do that if this should be set by the `toggleDatesSelectVisibility` function?